### PR TITLE
Simple MaskedArray implementation

### PR DESCRIPTION
**Pull request by halungge**
(FIX) poor man s implementation of a Masked Array as numpy.ma has no correspondence in cupy. 

Additional fixes:
- add missing host to device copy for serialbox
- fix imports

### CODE REVIEW

File: model/common/src/icon4py/model/common/decomposition/definitions.py [hunk 1]
```diff
@@ -59,21 +59,37 @@ def __call__(self):
             self._counter = self._counter + 1
         return next_id
 
+@dataclasses.dataclass(frozen = True)
+class MaskedArray:
+    data: np.ndarray
+    mask: np.ndarray
+    
+    def __post__init__(self, ):
```
Comment (OngChia) [inline model/common/src/icon4py/model/common/decomposition/definitions.py:67]: It looks like it has an extra _ between post and init

**Review by OngChia — APPROVED** [review]
Thanks a lot for the fix! Only one possible typo.

File: model/common/src/icon4py/model/common/decomposition/definitions.py [hunk 2]
```diff
@@ -93,7 +109,7 @@ def local_index(self, dim: Dimension, entry_type: EntryType = EntryType.ALL):
                 return index[mask]
 
     def _to_local_index(self, dim):
-        data = ma.getdata(self._global_index[dim], subok=False)
+        data = self._global_index[dim].data
         assert data.ndim == 1
         return np.arange(data.shape[0])
```
Comment (OngChia) [inline model/common/src/icon4py/model/common/decomposition/definitions.py:114]: xp?

File: model/common/src/icon4py/model/common/decomposition/definitions.py [hunk 3]
```diff
@@ -8,14 +8,14 @@
 
 from __future__ import annotations
 
+import dataclasses
 import functools
 import logging
 from dataclasses import dataclass
 from enum import IntEnum
```
Comment (OngChia) [inline model/common/src/icon4py/model/common/decomposition/definitions.py:15]: import statement

File: model/common/src/icon4py/model/common/decomposition/definitions.py [hunk 4]
```diff
@@ -59,21 +59,37 @@ def __call__(self):
...
+        self._global_index:dict[Dimension, MaskedArray] = {}
+        self._klevels = klevels
+        
     class EntryType(IntEnum):
         ALL = 0
         OWNED = 1
         HALO = 2
 
     @builder.builder
     def with_dimension(self, dim: Dimension, global_index: np.ndarray, owner_mask: np.ndarray):
```
[the remark this answers is not in the document]
Comment (halungge) [reply model/common/src/icon4py/model/common/decomposition/definitions.py:88]: samve

File: model/common/src/icon4py/model/common/decomposition/definitions.py (hunk 1 above)
Comment (halungge) [reply model/common/src/icon4py/model/common/decomposition/definitions.py:67]: thanks

**Pull request merged**